internal/buffer/pool_test.go

```diff
@@ -0,0 +1,37 @@
+package buffer
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuffers(t *testing.T) {
+	var wg sync.WaitGroup
+	for g := 0; g < 10; g++ {
+		wg.Add(1)
+		go func() {
+			for i := 0; i < 100; i++ {
+				buf := Get()
+				assert.Zero(t, buf.Len(), "Expected truncated buffer")
+
+				b := getRandBytes()
+				buf.Write(b)
```

Comment:
we probably want to fail if `buf.Write` fails with an error (even though it's probably impossible) just for the sake of `errcheck`